sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -4603,6 +4603,14 @@ object SQLConf {
     .booleanConf
     .createWithDefault(true)
 
+  val LEGACY_NO_CHAR_PADDING_IN_PREDICATE = buildConf("spark.sql.legacy.noCharPaddingInPredicate")
```

Comment:
Because the legacy behavior is always padding, how about `LEGACY_CHAR_PADDING_IN_PREDICATE` and change the default value to true ?

Reply:
legacy behavior is no padding.

Reply:
`However, there is a bug in Spark that we always pad the string literal when comparing CHAR type and STRING literals, which assumes the CHAR type columns are always padded, either on the write side or read side.`
I'm a bit confused.

Reply:
Let me make it clear. It was talking about we people disabling `spark.sql.readSideCharPadding`

Reply:
Got it. Thank you.